apps/src/templates/teacherDashboard/teacherSectionsRedux.js

```diff
@@ -164,22 +177,63 @@ export const editSectionLoginType = (sectionId, loginType) => dispatch => {
   return dispatch(finishEditingSection());
 };
 
-export const asyncLoadSectionData = () => (dispatch) => {
-  dispatch({type: ASYNC_LOAD_BEGIN});
-  return Promise.all([
-    fetchJSON('/dashboardapi/sections'),
-    fetchJSON('/dashboardapi/courses'),
-    fetchJSON('/v2/sections/valid_scripts')
-  ]).then(([sections, validCourses, validScripts]) => {
-    dispatch(setValidAssignments(validCourses, validScripts));
-    dispatch(setSections(sections));
-  }).catch(err => {
-    console.error(err.message);
-  }).then(() => {
-    dispatch({type: ASYNC_LOAD_END});
+export const updateShareSetting = (sectionId, shareSetting) => (dispatch, getState) => {
+  dispatch({type: UPDATE_SHARING_REQUEST});
+  return new Promise((resolve, reject) => {
+    $.ajax({
+      url: `dashboardapi/sections/${sectionId}/update_sharing_disabled`,
+      method: 'POST',
+      contentType: 'application/json;charset=UTF-8',
+      data: JSON.stringify({sharing_disabled: shareSetting}),
+    }).done(result => {
+      dispatch({
+        type: UPDATE_SHARING_SUCCESS,
+        sectionId: sectionId,
+        serverSectionShareSetting: result.sharing_disabled,
+        serverStudents: result.students
+      });
+      resolve();
+    }).fail((jqXhr, status) => {
+      dispatch({type: UPDATE_SHARING_FAILURE});
+      reject(status);
+    });
   });
 };
 
+export const asyncLoadSectionData = (id) => (dispatch) => {
```

Comment:
sometimes this is called without a section ID and since it's chained, I'm not sure how to write this more efficiently.